base/dcp_dest.go

```diff
@@ -188,13 +191,19 @@ func (d *DCPDest) OpaqueSet(partition string, value []byte) error {
 	return nil
 }
 
+// Rollback is required by cbgt.Dest interface but will not work when called by Sync Gateway as we need additional information to perform a rollback. Due to the design of cbgt.Dest this will not be called without a programming error.
 func (d *DCPDest) Rollback(partition string, rollbackSeq uint64) error {
-	return d.rollback(partitionToVbNo(d.loggingCtx, partition), rollbackSeq)
+	panic("Only RollbackEx should be called, this function is required to be implmented by cbgt.Dest interface. This function does not provide Sync Gateway with enough information to rollback.")
```

Comment:
I'd prefer to avoid panicking and continue to log a warning (or possibly error, if you think it's that severe).

Reply:
This error will result in the feed not running, turned this into a warning.